tests/std/tests/P2467R1_exclusive_mode_fstreams/test.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <filesystem>
+#include <fstream>
+
+namespace fs = std::filesystem;
+
+#define FILENAME L"test.dat"
+
+struct CreateTempFile {
+    fs::path m_path;
+    CreateTempFile(const fs::path& path) : m_path(path) {
+        std::ofstream c{path};
+    }
+
+    ~CreateTempFile() {
+        fs::remove(m_path);
+    }
+};
+
+void test_file_create(std::ios_base::openmode mode) {

```

Comment:
Also meganit (I don't feel bad since I've committed myself to applying these cleanups instead of making you do it) the parameters to these functions could be `const` to ensure we don't accidentally modify them locally.